### PR TITLE
Added initial TelUri support

### PR DESCRIPTION
**Pull request by dmnava**
Hi this is an initial draft of adding tel uri support

### CODE REVIEW

**Comment by aboutsip** [comment]
Awesome! Just a few things:
1. No tabs please, spaces only for indentation.
2. Could you add some more unit tests for failure scenarios as well?

File: pkts-sip/src/main/java/io/pkts/packet/sip/address/impl/TelURIImpl.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ *
+ */
+public class TelURIImpl extends URIImpl implements TelURI {
+	
+	private final boolean isGlobal;
+	private final Buffer phoneNumber;
+	private final Buffer headers;
+	private final ParametersSupport paramsSupport;
+	
+	public TelURIImpl(boolean isGlobal, Buffer phoneNumber, Buffer headers, Buffer original) {
```
Comment (aboutsip) [inline pkts-sip/src/main/java/io/pkts/packet/sip/address/impl/TelURIImpl.java]: nit; add final to these a well

File: pkts-sip/src/test/java/io/pkts/packet/sip/address/impl/TelURIImplTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import io.pkts.buffer.Buffer;
+import io.pkts.buffer.Buffers;
+import io.pkts.packet.sip.SipParseException;
+import io.pkts.packet.sip.address.TelURI;
+
+import static org.junit.Assert.*;
+
+import java.io.IOException;
+
+public class TelURIImplTest {
```
Comment (aboutsip) [inline pkts-sip/src/test/java/io/pkts/packet/sip/address/impl/TelURIImplTest.java]: Could you add some more tests around bad tel-uris?

Also, could you add some tests where you still have stuff left in the buffer and ensure whatever content that is left is indeed still there. I.e. 

```
Buffer buffer = Buffers.wrap("tel:+34600100100\r\nsome_other_stuff")
TelURI telUri = TelURI.frame(buffer);
assertEquals("34600100100", telUri.getPhoneNumber().toString());
assertTrue(telUri.isGlobal());
assertThat(buffer.toString(), is("some_other_stuff");```

File: pkts-sip/src/main/java/io/pkts/packet/sip/address/impl/TelURIImpl.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	@Override
+	public URI clone() {
+		return new TelURIImpl(isGlobal, phoneNumber, headers, getRawURI());
+	}
+
+	@Override
+	public Builder<TelURI> copy() {
+		// TODO Not yet implemented
```
Comment (aboutsip) [inline pkts-sip/src/main/java/io/pkts/packet/sip/address/impl/TelURIImpl.java]: Will you get to this one now as well?

**Comment by dmnava** [comment]
Hi,

I have implemented TelURI Builder as well as added some more unit tests and fixed some bugs

Daniel

**Comment by aboutsip** [comment]
Thanks! Could you just squash your commits into one, force push it and then I'll merge it...

**Comment by dmnava** [comment]
Done the squash and forced commit

**Pull request merged**